src/transport_ofi.h

```diff
@@ -1285,4 +1289,29 @@ uint64_t shmem_transport_pcntr_get_completed_target(shmem_transport_ctx_t *ctx)
     return cnt;
 }
 
+static inline
+void shmem_transport_pcntr_get_all(shmem_transport_ctx_t *ctx, shmem_pcntr_t *pcntr)
+{
+    SHMEM_TRANSPORT_OFI_CTX_LOCK(ctx);
+
+    pcntr->completed_put = fi_cntr_read(ctx->put_cntr);
+    if (ctx->options & SHMEMX_CTX_BOUNCE_BUFFER) {
+        SHMEM_TRANSPORT_OFI_CTX_BB_LOCK(ctx);
+        pcntr->completed_put += ctx->completed_bb_cntr;
+        SHMEM_TRANSPORT_OFI_CTX_BB_UNLOCK(ctx);
+    }
+    pcntr->completed_get = fi_cntr_read(ctx->get_cntr);
+
+    pcntr->pending_put = SHMEM_TRANSPORT_OFI_CNTR_READ(&ctx->pending_put_cntr);
+    pcntr->pending_get = SHMEM_TRANSPORT_OFI_CNTR_READ(&ctx->pending_get_cntr);
+    if (ctx->options & SHMEMX_CTX_BOUNCE_BUFFER) {
+        SHMEM_TRANSPORT_OFI_CTX_BB_LOCK(ctx);
```

Comment:
Can we merge this critical section on the BB lock with the one above?

Reply:
It was like that before. I am not sure now why I separated them :( .. Anyway, will merge.